### PR TITLE
Support png buttons

### PR DESCRIPTION
**Pull request by johanmalm**
TODO

- [x] Documentation
- [x] Test with big/small buttons and make sure the scale sensibly
- [x] Test with width != height and make sure they scale sensibly

Some review in the meantime would be appreciated

### CODE REVIEW

**Comment by 01micko** [comment]
Works ok.

png-icons


Just for the record, I figured I needed 6 buttons (well 8 really for  menu-[in]active.png)
- close-[in]active.png
- iconify-[in]active.png
- max-[in]active.png

<details>
<summary>I converted the following svg images</summary>

close

```svg
<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" version="1.1" height="100" width="100">

  <rect style="fill:none;stroke:#FFFFFF;stroke-width:3" width="94" height="80" x="3" y="10"/>
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 32 35 68 70" />
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 68 35 32 70" />
</svg>
```

iconify

```svg
<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" version="1.1" height="100" width="100">

  <rect style="fill:none;stroke:#FFFFFF;stroke-width:3" width="94" height="80" x="3" y="10"/>
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 18 30 82 30" />
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 50 83 50 30" />
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 25 55 50 80" />
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 75 55 50 80" />
</svg>
```

max

```svg
<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" version="1.1" height="100" width="100">

  <rect style="fill:none;stroke:#FFFFFF;stroke-width:3" width="94" height="80" x="3" y="10"/>
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 18 75 82 75" />
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 50 75 50 22" />
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 25 50 50 25" />
  <path style="stroke:#FFFFFF;stroke-width:8" d="M 75 50 50 25" />
</svg>
```

</summary>

File: src/button-png.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <wlr/util/log.h>
+#include <png.h>
+#include <sys/stat.h>
+#include "common/dir.h"
+#include "buffer.h"
+#include "button-png.h"
+#include "labwc.h"
+#include "theme.h"
```
Comment (Consolatis) [inline src/button-png.c:17]: .. not being ordered alphabetically.

**Review by Consolatis — COMMENTED** [review]
This looks great!

Other than the small issues from `scripts/check`, my nitpick-o-meter only stumbled over the includes

**Comment by johanmalm** [comment]
Ready to go from my perspective. Will merge tomorrow unless there any final thoughts.

Have tested big small icons.
Big icons scale down to max available space. Small icons retain original size.

File: docs/labwc-theme.5.scd [hunk 2]
```diff
@@ -18,7 +18,8 @@ searched for in the following order:
 Choosing a theme is done by editing the <name> key in the <theme> section of
 the rc.xml configuration file (labwc-config(5)).
 
-A theme consists of a themerc file and optionally some xbm icons.
+A theme consists of a themerc file and optionally some titlebar icons (referred
+to use buttons).
```
Comment (Consolatis) [inline docs/labwc-theme.5.scd:22]: `s/use/as/` ?

Other than that, LGTM.

Comment (johanmalm) [reply docs/labwc-theme.5.scd]: Thanks

**Comment by Consolatis** [comment]
@narrat Friendly ping about a new -master dep: libpng
Might already be satisfied indirectly by another dep like cairo though.

**Comment by Narrat** [comment]
Thanks for the heads up.
It is indeed indirectly covered by cairo, but as labwc directly uses the library I will add it to the package. Just in case.

**Pull request merged**